src/wallet/test/wallet_tests.cpp

```diff
@@ -337,37 +343,38 @@ BOOST_AUTO_TEST_CASE(LoadReceiveRequests)
     BOOST_CHECK_EQUAL(values[1], "val_rr1");
 }
 
-// Test some watch-only wallet methods by the procedure of loading (LoadWatchOnly),
+// Test some watch-only LegacyScriptPubKeyMan methods by the procedure of loading (LoadWatchOnly),
 // checking (HaveWatchOnly), getting (GetWatchPubKey) and removing (RemoveWatchOnly) a
 // given PubKey, resp. its corresponding P2PK Script. Results of the the impact on
 // the address -> PubKey map is dependent on whether the PubKey is a point on the curve
-static void TestWatchOnlyPubKey(CWallet& wallet, const CPubKey& add_pubkey)
+static void TestWatchOnlyPubKey(LegacyScriptPubKeyMan* spk_man, const CPubKey& add_pubkey)
 {
     CScript p2pk = GetScriptForRawPubKey(add_pubkey);
     CKeyID add_address = add_pubkey.GetID();
     CPubKey found_pubkey;
-    LOCK(wallet.cs_wallet);
+    LOCK2(spk_man->cs_wallet, spk_man->cs_KeyStore);
```

Comment:
In commit "Refactor: Split up CWallet and LegacyScriptPubKeyMan and classes" (194227b1173b0dd9a51d24e245ddaa601d36a109)

Extra lock here seems not ideal. This commit is not supposed to change behavior in any way. Other places in this commit I just added AssertLockHelds where needed to avoid clang lock warnings, but here I'm not even seeing that is necessary.

Reply:
It was failing the tests without it.

Reply:
> It was failing the tests without it.

So this is definitely a bug. There can’t be any changes in behavior in this PR. All it is supposed do is add wrapper functions and variable aliases. If any side effects are present from this, it means something has gone wrong.

Any case I reproduced the problem and found it was limited to tests and caused by me pulling in some test-only LOCK changes that didn't belong here in a bad rebase. This PR can be fixed by just updating the tip from 1c49a20cb10f2d48cf23bb364d586a9c1e86d5ec to f201ba59ffd2e071a36a688b80d2cff9a9c44bb2 ([compare](https://github.com/bitcoin/bitcoin/compare/1c49a20cb10f2d48cf23bb364d586a9c1e86d5ec..f201ba59ffd2e071a36a688b80d2cff9a9c44bb2)) from my updated [`pr/keyman`](https://github.com/ryanofsky/bitcoin/commits/pr/keyman) branch

(You can also reset #17261 to the updated [`pr/keyman`](https://github.com/ryanofsky/bitcoin/commits/pr/keyman) branch with no loss since I pulled in your updated SigningProvider commit there.)

Reply:
Done